third_party/blink/renderer/modules/mediasource/media_source.cc

```diff
@@ -492,13 +498,51 @@ bool MediaSource::isTypeSupported(ExecutionContext* context,
   bool result = IsTypeSupportedInternal(
       context, type, true /* Require fully specified mime and codecs */);
   DVLOG(2) << __func__ << "(" << type << ") -> " << (result ? "true" : "false");
+#if BUILDFLAG(IS_COBALT)
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  LOG(INFO) << __func__ << "(" << type << ") -> "
+            << (result ? "true" : "false");
+#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
+#endif // BUILDFLAG(IS_COBALT)
   return result;
 }
 
 // static
 bool MediaSource::IsTypeSupportedInternal(ExecutionContext* context,
                                           const String& type,
                                           bool enforce_codec_specificity) {
+
+#if BUILDFLAG(IS_COBALT)
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
```

Comment:
nit: I'd do
```
#if BUILDFLAG(USE_STARBOARD_MEDIA)
#if !BUILDFLAG(IS_COBALT)
#error ...
#endif
```

but in all likelihood this should be asserted in the BUILD.gn files,  
or perhaps in "build/buildflag.h".

(This is just a high level comment, not a review :)